transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/logging/Logging.java

```diff
@@ -14,13 +14,26 @@
 
 package com.google.android.datatransport.runtime.logging;
 
+import android.os.Build;
 import android.util.Log;
 
 public final class Logging {
   private Logging() {}
 
   private static String getTag(String tag) {
-    return "TransportRuntime." + tag;
+    String prefix = "TRuntime.";
+    return prefix + concatTag(prefix, tag);
+  }
+
+  private static String concatTag(String prefix, String tag){
+    int loggableLimit = 23;
+    int remainingSpace = loggableLimit - prefix.length();
+
+    if(android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.N)
```

Comment:
to prevent an unnecessary function call, I'd make the function always cap to max 23, and move the `os.Build` check to `getTag`.